submitter/relayer/relayer.go

```diff
@@ -356,106 +321,100 @@ func (rl *Relayer) ChainTwoTxAndSend(
-	if err != nil {
-		return nil, fmt.Errorf("failed to get change address: %w", err)
-	}
-	rl.logger.Debugf("Got a change address %v", changeAddr.String())
-	changeScript, err := txscript.PayToAddrScript(changeAddr)
+	changePosition := 1 // must declare here as you cannot take address of const needed bellow
+	feeRate := btcutil.Amount(rl.getFeeRate()).ToBTC()
+	rawTxResult, err := rl.BTCWallet.FundRawTransaction(tx, btcjson.FundRawTransactionOpts{
+		FeeRate:        &feeRate,
+		ChangePosition: &changePosition,
+	}, nil)
 	if err != nil {
 		return nil, err
 	}
-	copiedTx := &wire.MsgTx{}
-	err = copier.Copy(copiedTx, tx)
+
+	rl.logger.Debugf("Building a BTC tx using %s with data %x", rawTxResult.Transaction.TxID(), data)
+
+	_, addresses, _, err := txscript.ExtractPkScriptAddrs(
+		rawTxResult.Transaction.TxOut[changePosition].PkScript,
+		rl.GetNetParams(),
+	)
+
 	if err != nil {
 		return nil, err
 	}
-	txSize, err := calculateTxVirtualSize(copiedTx, utxo, changeScript)
+
+	if len(addresses) == 0 {
+		return nil, errors.New("no change address found")
+	}
+
+	changeAddr := addresses[0]
+	rl.logger.Debugf("Got a change address %v", changeAddr.String())
+
+	txSize, err := calculateTxVirtualSize(rawTxResult.Transaction)
 	if err != nil {
 		return nil, err
 	}
+
+	changeAmount := btcutil.Amount(rawTxResult.Transaction.TxOut[changePosition].Value)
 	minRelayFee := rl.calcMinRelayFee(txSize)
-	if utxo.Amount < minRelayFee {
-		return nil, fmt.Errorf("the value of the utxo is not sufficient for relaying the tx. Require: %v. Have: %v", minRelayFee, utxo.Amount)
+
+	if changeAmount < minRelayFee {
+		return nil, fmt.Errorf("the value of the utxo is not sufficient for relaying the tx. Require: %v. Have: %v", minRelayFee, changeAmount)
 	}
-	txFee := rl.getFeeRate().FeeForVSize(txSize)
+
+	txFee := rawTxResult.Fee
 	// ensuring the tx fee is not lower than the minimum relay fee
 	if txFee < minRelayFee {
 		txFee = minRelayFee
 	}
 	// ensuring the tx fee is not higher than the utxo value
-	if utxo.Amount < txFee {
-		return nil, fmt.Errorf("the value of the utxo is not sufficient for paying the calculated fee of the tx. Calculated: %v. Have: %v", txFee, utxo.Amount)
+	if changeAmount < txFee {
+		return nil, fmt.Errorf("the value of the utxo is not sufficient for paying the calculated fee of the tx. Calculated: %v. Have: %v", txFee, changeAmount)
 	}
-	change := utxo.Amount - txFee
-	tx.AddTxOut(wire.NewTxOut(int64(change), changeScript))
 
 	// sign tx
-	tx, err = rl.dumpPrivKeyAndSignTx(tx, utxo)
+	tx, err = rl.signTx(rawTxResult.Transaction)
 	if err != nil {
 		return nil, fmt.Errorf("failed to sign tx: %w", err)
 	}
 
 	// serialization
 	var signedTxBytes bytes.Buffer
-	err = tx.Serialize(&signedTxBytes)
-	if err != nil {
+	if err := tx.Serialize(&signedTxBytes); err != nil {
 		return nil, err
 	}
 
+	change := changeAmount - txFee
+
 	rl.logger.Debugf("Successfully composed a BTC tx with balance of input: %v, "+
```

Comment:
We probably need to remove the balance of input as we don't specify it anymore. And we need to make it a single line as otherwise the values are not well translated